From c16d7066c7abfe53315131a6a58203074615b615 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 26 Jul 2022 16:07:07 +0300
Subject: [PATCH] [SW-97612] gaudi2: fix sleep macro definition

The sleep macro definition should be inside the !_ASSEMBLY__ macro
scope.
Move the sleep macro to a proper location.
While at this, move the ifndef __ASSEMBLY__ scope to the end of the
file, so it will not be a block in the middle - just for better
organization.

Change-Id: Ib231fc82a8bb8e736ea87cd84052799ac17adae1
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index ffa8cac..8c5ca92 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -112,16 +112,16 @@
 #define ARC_ADDR_MASK		0x7FFFFFFull
 #define ARC_SDRAM_ADDR_MASK	0x3FFFFFFull
 
-#define sleep()	__asm__ __volatile__("sleep\n\t" : : )
-
-#ifndef __ASSEMBLY__
-void init_secondary(void);
-extern void secondary_entry(void);
-#endif
-
 #define TS_UNCALIB_A_G_COEFF	81100
 #define TS_UNCALIB_B_H_COEFF	255200
 #define TS_CALIB_A_G_COEFF	82600
 #define TS_CALIB_B_H_COEFF	262600
 
+#ifndef __ASSEMBLY__
+#define sleep()	__asm__ __volatile__("sleep\n\t" : : )
+
+void init_secondary(void);
+extern void secondary_entry(void);
+#endif /* __ASSEMBLY__ */
+
 #endif /* COMMON_GAUDI2_ARC_GNU_H_ */
